src/Console/ConfigurationResolver.php

```diff
@@ -554,7 +554,7 @@ private function computeConfigFiles(): array
             $configDir = $path[0];
         } else {
             $dirName = pathinfo($path[0], PATHINFO_DIRNAME);
-            $configDir = $dirName ?: $path[0];
+            $configDir = is_dir($dirName) ? $dirName : $path[0];
```

Comment:
we already took PATHINFO_DIRNAME line above, why we need to check if returned dirname is dir ?

Reply:
```php
$paths = [__FILE__, '/non/existing/path/foo.php'];

foreach ($paths as $path) {
    $dirName = pathinfo($path, PATHINFO_DIRNAME);
    $configDir = is_dir($dirName) ? $dirName : $path;

    var_dump($dirName, is_dir($dirName), $configDir);
}
```

```shell
$ php /Volumes/Projects/~Github/PHP-CS-Fixer/PHP-CS-Fixer/reproducer.php

string(51) "/Volumes/Projects/~Github/PHP-CS-Fixer/PHP-CS-Fixer"
bool(true)
string(51) "/Volumes/Projects/~Github/PHP-CS-Fixer/PHP-CS-Fixer"
string(18) "/non/existing/path"
bool(false)
string(26) "/non/existing/path/foo.php"
```

`PATHINFO_DIRNAME` only determines which part is parsed and returned, it does not enforce validity. With `$dirName ?: $path[0]` PHPStan reports `Short ternary operator is not allowed. Use null coalesce operator if applicable or consider using long ternary.`. Changing it to `$dirName ? $dirName : $path[0]` leads to `Only booleans are allowed in a ternary operator condition, string given`, hence the explicit check with `is_dir()`.

Reply:
thank you :thanks: